app/api/upload-csv/route.ts

```
@@ -0,0 +1,180 @@
+    // Validate records presence
+    if (!records || records.length === 0) {
+      console.log('No records found in CSV');
+      return NextResponse.json(
+        { success: false, error: 'CSV file is empty' },
+        { status: 400 }
+      );
+    }
+
+    // Validate required fields presence
+    const missingFields = REQUIRED_FIELDS.filter(
+      field => !records[0].hasOwnProperty(field)
```

Biome (1.9.4)

[error] Do not access Object.prototype method 'hasOwnProperty' from target object.

It's recommended using Object.hasOwn() instead of using Object.hasOwnProperty().
See MDN web docs for more details.

(lint/suspicious/noPrototypeBuiltins)